packages/app/src/components/DBInfraPanel.tsx

```diff
@@ -0,0 +1,277 @@
+                  series: [
+                    {
+                      type: 'time',
+                      where,
+                      groupBy: [],
+                      aggFn: 'avg',
+                      field: `${fieldPrefix}filesystem.available - Gauge`,
+                      table: 'metrics',
+                      numberFormat: K8S_FILESYSTEM_NUMBER_FORMAT,
+                    },
+                  ],
+                },
+                {
+                  metric: metricSource,
+                },
+              )}
+              showDisplaySwitcher={false}
+              logReferenceTimestamp={timestamp / 1000}
+            />
+          </Card.Section>
+        </Card>
+      </SimpleGrid>
+    </div>
+  );
+};
+
+export default ({
+  rowData,
+  rowId,
+  source,
+}: {
+  rowData?: Record<string, any>;
+  rowId: string | undefined | null;
+  source: TSource;
+}) => {
+  const { data: metricSource } = useSource({ id: source.metricSourceId });
+
+  const podUid = rowData?.__hdx_resource_attributes['k8s.pod.uid'];
+  const nodeName = rowData?.__hdx_resource_attributes['k8s.node.name'];
+
+  const timestamp = new Date(rowData?.__hdx_timestamp).getTime();
+
+  return (
+    <Stack my="md" gap={40}>
+      {podUid && (
+        <div>
+          {metricSource && (
+            <InfraSubpanelGroup
+              title="Pod"
+              where={`${metricSource.resourceAttributesExpression}.k8s.pod.uid:"${podUid}"`}
+              fieldPrefix="k8s.pod."
+              timestamp={timestamp}
+              metricSource={metricSource}
+            />
+          )}
+          {source && (
+            <Card p="md" mt="xl">
+              <Card.Section p="md" py="xs" withBorder>
+                Pod Timeline
+              </Card.Section>
+              <Card.Section>
+                <ScrollArea
+                  viewportProps={{
+                    style: { maxHeight: 280 },
+                  }}
+                >
+                  <Box p="md" py="sm">
+                    <KubeTimeline
+                      logSource={source}
+                      q={`${getEventBody(source)}.object.regarding.uid:"${podUid}"`}
+                      dateRange={[
+                        sub(new Date(timestamp), { days: 1 }),
+                        add(new Date(timestamp), { days: 1 }),
+                      ]}
+                      anchorEvent={{
+                        label: <div className="text-success">This Event</div>,
+                        timestamp: new Date(timestamp).toISOString(),
+                      }}
+                    />
+                  </Box>
+                </ScrollArea>
+              </Card.Section>
+            </Card>
+          )}
+        </div>
+      )}
+      {nodeName && metricSource && (
+        <InfraSubpanelGroup
+          metricSource={metricSource}
+          title="Node"
+          where={`${metricSource.resourceAttributesExpression}.k8s.node.name:"${nodeName}"`}
```

Comment:
this assumes the expression is a column name, but it can be any arbitrary sql expression and this will break